numpy/_core/tests/test_deprecations.py

```diff
@@ -454,3 +454,17 @@ def test_deprecated(self):
                 struct_ufunc.add_triplet, "new docs"
             )
         )
+
+
+def test_deprecated_T_non_2dim():
+    # Deprecated in Numpy 2.3, 2025-04
+    with pytest.warns(UserWarning, match="In the future `.T` property for "
```

Comment:
Please use `assert_deprecated` as all other tests in this file.  That ensures that the error path is also tested, even if that is rather trivial here.

Reply:
I didn't use `assert_deprecated` because it's missing `match=` parameter to actually match the warning message, but I can use it instead.

Reply:
> because it's missing match= parameter to actually match the warning message

Seems reasonable to add it, assuming that's easy.

Reply:
IIRC it does, but takes the message from the class, but not sure. And yeah, I don't actually care about using it but I do care abot a habit of testing the error path.

Reply:
Thanks for an explanation! I updated the message as you suggested.

Reply:
> > because it's missing match= parameter to actually match the warning message
> 
> Seems reasonable to add it, assuming that's easy.

I like this idea - added `msg_patterns` argument to `assert_deprecated`.